mirrord-agent/src/file.rs

```diff
@@ -198,6 +188,52 @@ impl FileManager {
             })
     }
 
+    #[tracing::instrument(level = "trace", skip(self))]
+    pub(crate) fn read_string(
+        &mut self,
+        fd: usize,
+        buffer_size: usize,
+    ) -> RemoteResult<ReadStringFileResponse> {
+        self.open_files
+            .get_mut(&fd)
+            .ok_or(ResponseError::NotFound(fd))
+            .and_then(|remote_file| {
+                if let RemoteFile::File(file) = remote_file {
+                    let mut reader = BufReader::new(std::io::Read::by_ref(file));
+                    let mut buffer = String::with_capacity(buffer_size);
+                    let read_result = reader
+                        .read_line(&mut buffer)
```

Comment:
This is a bit ugly (conceptually, not the implementation). I'd document it carefully as *I think* I understand what's going on, but better safe than sorry.
(I understood that the "best" way to read line in Rust is using BufReader, which might advance the file offset further than it'd have been if we just used fgets due to the buffering.. R_R)

Reply:
It's less about being the best, and more about `read_line` not being implemented for `File`. Added some comments there, do they make things clearer?

Reply:
I think a short paragraph before the implementation is better than inline comments.

Reply:
Added a proper doc for this. Left some of the inline comments (reduced the text as some was moved to the doc). What do you think?